demo2/products/productsController.js

```diff
@@ -0,0 +1,85 @@
+const productsModel = require('./productsModel');
+const productsView = require('./productsView');
+const NotFound = require('../common/errors/notFound');
+
+class ProductsController {
+
+    getAllProducts = async (req, res, next) => {
+        try {
+            const products = await productsModel.getProducts('products');
+            productsView.productsView(res, products);
+        }
+        catch {
+            next(new NotFound);
+        }
+        
+    }
+
+
+    getProduct = async (req, res,next) => {
+        const { id } = req.params;
+        try {
+            const product = await productsModel.getProduct(id);
+            productsView.productView(res, product);
+        }
+        catch {
+            next(new NotFound);
+        }
```

Comment:
1. new NotFound('Error message')
2. catch(e) {next(e)}
3. you can wrap the whole thing with asynchandler